### DISCUSSION TITLE
8-color/3-bit fallback (for tty)

### DISCUSSION THREAD

**Discussion by MithicSpirit** [catppuccin/nvim#759] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Using the catppuccin theme makes neovim terrible to use in a regular TTY, since everything is black and white.

**Describe the solution you'd like**
I think that it would be helpful to provide an 8-color/3-bit fallback for such situations so that it's still usable, even if it doesn't look as good as the normal theme.

**Describe alternatives you've considered**
I could also conditionally set up catppuccin or another theme, depending on my terminal's capabilities.

**Comment by vollowx** [comment, marked as answer]
Try the following code snippet, since the default colorscheme supports TTY best.

```lua
-- At where you set the colorscheme

vim.g.has_ui = #vim.api.nvim_list_uis() > 0
vim.g.has_gui = vim.g.has_ui and (vim.env.DISPLAY ~= nil or vim.env.WAYLAND_DISPLAY ~= nil)

if not vim.g.has_gui then
  if vim.g.has_ui then
    vim.cmd.colorscheme('default')
  end
  return
end

vim.cmd.colorscheme({
  args = { 'catppuccin' },
  mods = { emsg_silent = true },
})
```
